.castor.php

```diff
@@ -0,0 +1,440 @@
+<?php
+
+use Castor\Attribute\AsContext;
+use Castor\Attribute\AsOption;
+use Castor\Attribute\AsTask;
+use Castor\Context;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Process\Exception\ExceptionInterface;
+use Symfony\Component\Process\ExecutableFinder;
+use Symfony\Component\Process\Process;
+
+use function Castor\exec;
+
+#[AsTask(description: 'Builds the infrastructure', namespace: 'infra', aliases: ['build'])]
+function build(Context $c)
+{
+    $command = [
+        'build',
+        '--build-arg', "USER_ID={$c['user_id']}",
+        '--build-arg', "PHP_VERSION={$c['php_version']}",
+    ];
+
+    return docker_compose($c, $command, withBuilder: true);
+}
+
+#[AsTask(description: 'Builds and starts the infrastructure', namespace: 'infra', aliases: ['up'])]
+function up(Context $c)
+{
+    $command = [
+        'up',
+        '--remove-orphans',
+        '--detach',
+    ];
+
+    return docker_compose($c, $command);
+}
+
+#[AsTask(description: 'Builds and starts the infrastructure, then install the application (composer, yarn, ...)')]
+function start(Context $c, SymfonyStyle $io)
+{
+    workers_stop($c);
+    generate_certificates($c, $io, false);
+    build($c);
+    up($c);
+    cache_clear($c);
+    install($c);
+    migrate($c);
+    workers_start($c);
+
+    $io->success('The stack is now up and running.');
+
+    about($c, $io);
+}
+
+#[AsTask(description: 'Installs the application (composer, yarn, ...)', namespace: 'app')]
+function install(Context $c)
+{
+    $basePath = sprintf('%s/%s', $c['root_dir'], $c['project_directory']);
+
+    if (is_file("{$basePath}/composer.json")) {
+        docker_compose_run($c, 'composer install -n --prefer-dist --optimize-autoloader');
+    }
+    if (is_file("{$basePath}/yarn.lock")) {
+        docker_compose_run($c, 'yarn');
```

Comment:
can the two previous tasks (composer and yarn install) be executed in parallel?

Reply:
In another PR?